tests/xtro-sharpie/SelectorCheck.cs

```diff
@@ -4,38 +4,20 @@
 // !missing-selector!
 //             if headers defines a selector for which we have no bindings
 //
-// !unknown-selector!
-//             if we have a selector that is not part of the header files
-//
 
 using System;
 using System.Collections.Generic;
 
 using Mono.Cecil;
 
 using Clang.Ast;
+using System.Linq;
 
 namespace Extrospection {
 
 	public class SelectorCheck : BaseVisitor {
-//		Dictionary<string,List<MethodDefinition>> exports = new Dictionary<string, List<MethodDefinition>> ();
-
-		// missing
-		//	-> it's not in the type or ancestor or their interface (protocols)
-		//	-> it's not in a category
-
-		// unknown
-		//	-> quick check (HashSet) to see if it's used anywhere
-		//	-> 
-
-		// duplicate
-		//	-> the selector is defined more than once for the same type
 
-		HashSet<string> known_selectors = new HashSet<string> ();
-
-		HashSet<string> qualified_selectors = new HashSet<string> ();
-
-		//Dictionary<TypeDefinition,HashSet<string>> type_exports = new Dictionary<TypeDefinition,HashSet<string>> ();
+		HashSet<(string MethodDefinition, Helpers.ArgumentSemantic ArgumentSemantic)> qualified_export_arguments = new HashSet<(string, Helpers.ArgumentSemantic)> ();
```

Comment:
you're now allocating a tuple for each entry - even if only properties actually needs the extra argument semantics

Reply:
depending on the numbers it might be cheaper to have a dictionary where you store argument semantics (and only when available) along with the `MethodDefinition`

Reply:
Can't this be `Dictionary<string, ArgumentSemantic>`? Or can you have multiple methods with the same name, but different argument semantics?

That would mean any lookup would be a hashed lookup, and not a sequential lookup.